content/documents/implementing-single-sign-on-with-google-idp.mdx

```diff
@@ -0,0 +1,413 @@
+await SecurityProfiles.SaveAssignment({
+  SecurityProfileID: "buyerProfile",
+  BuyerID: "buyer1"
+})
+.catch((err:OrderCloudError) => console.log(err));`,
+csharp: `using OrderCloud.SDK;\n
+var client = new OrderCloudClient(...);\n
+await client.AuthenticateAsync();\n
+await client.SecurityProfiles.SaveAssignmentAsync(new SecurityProfileAssignment {
+  SecurityProfileID = "buyerProfile",
+  BuyerID = "buyer1"
+});`}}
+/>
+
+### 4 Create an API client
+<CodeExample
+  content={{
+    http: `POST https://sandboxapi.ordercloud.io/v1/apiclients HTTP/1.1
+Authorization: Bearer INSERT_ACCESS_TOKEN_HERE
+Content-Type: application/json; charset=UTF-8;\n
+{
+    "AccessTokenDuration": 600,
+    "Active": true,
+    "AppName": "Buyer Client",
+    "RefreshTokenDuration": 43200,
+    "AllowAnyBuyer": true,
+    "AllowSeller": true
+}`,
+javascript: `import { Tokens, ApiClients } from "ordercloud-javascript-sdk";\n
+Tokens.Set("INSERT_ACCESS_TOKEN_HERE")\n
+ApiClients.Create({
+  AccessTokenDuration: 600,
+  Active: true,
+  AppName: "Buyer Client",
+  RefreshTokenDuration: 43200,
+  AllowAnyBuyer: true,
+  AllowSeller: true
+})
+.then((response) => {
+    console.log(response);
+})
+.catch(err => console.log(err));\n
+})`,
+typescript: `import { Tokens, ApiClients, ApiClient, OrderCloudError } from "ordercloud-javascript-sdk";\n
+Tokens.Set("INSERT_ACCESS_TOKEN_HERE")\n
+const apiClient: ApiClient = await ApiClients.Create({
+  AccessTokenDuration: 600,
+  Active: true,
+  AppName: "Buyer Client",
+  RefreshTokenDuration: 43200,
+  AllowAnyBuyer: true,
+  AllowSeller: true
+})
+.catch((err:OrderCloudError) => console.log(err));
+console.log(apiClient);`,
+csharp: `using OrderCloud.SDK;\n
+var client = new OrderCloudClient(...);\n
+await client.AuthenticateAsync();\n
+ApiClient response = await client.ApiClients.Create(new ApiClient {
+  AccessTokenDuration = 600,
+  Active = true,
+  AppName = "Buyer Client",
+  RefreshTokenDuration = 43200,
+  AllowAnyBuyer = true,
+  AllowSeller = true
+});`}}
+/>
+
+Make sure to record the ID from the response. You will need it for <Link to="/knowledge-base/implementing-single-sign-on-with-google-idp#4-create-a-new-openid-connect">step #2.4</Link>
+
+## 2. Configuring OIDC (OpenIDConnect) via OrderCloud
+
+### 1. Start Ngrok
+
+We'll need a publicly available endpoint. We can use a tool called [ngrok](https://ngrok.com/) to let us do this locally without having to deploy anything. After installing ngrok run the command `ngrok http 4451`. This tells ngrok to expose our endpoint (not yet running) on http://localhost:4451 to two public endpoints. After running the command copy either one of those URLs and record it, we'll need it for <Link to="/knowledge-base/implementing-single-sign-on-with-google-idp#2-create-the-openid-connect-integration-event">step #2.2</Link>
+
+> We recommend to keep ngrok running, restarting it will generate unique public endpoints and require you to update your configuration in OrderCloud.
```

Comment:
We recommend to keep ngrok running.* Restarting it will generate unique public endpoints and require you to update your configuration in OrderCloud.